### PR TITLE
Feature: Create an echo command.

### LINKED ISSUE #1895: [FEAT] - `/misc echo` command
### What feature would you like to see?

We should add a command which allows members to send a message from Becca's account.

- This should be locked to moderators (we can use the `Manage Messages` permission for this check)
- The message should be sent in the channel the command is called in.
- The interaction response should be ephemeral.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by a2937**
# Pull Request

## Description

It creates an command that allows members to send a message from Becca's account. 

## Related Issue
Closes #1895

## Documentation

For _any_ version updates, please verify if the [documentation page](https://docs.beccalyria.com?utm_source=github&utm_medium=pr-template) needs an update. If it does, please [create an issue there](https://github.com/BeccaLyria/discord-documentation/issues/new?assignees=nhcarrigan&labels=%F0%9F%9A%A6+status%3A+awaiting+triage&template=update.md&title=%5BUPDATE%5D) to ensure it is not forgotten.

- [ ] My contribution does NOT require a documentation update.
- [X] My contribution DOES require a documentation update.

Note: It's not been thoroughly tested because I was having some unexpected difficulties with Prisma. Additionally it appears that setting the permission for the command is impossible without adjusting the permissions for *all* the sub commands.

### CODE REVIEW

File: src/commands/subcommands/misc/handleEcho.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { CommandHandler } from "../../../interfaces/commands/CommandHandler";
+import { errorEmbedGenerator } from "../../../modules/commands/errorEmbedGenerator";
+import { beccaErrorHandler } from "../../../utils/beccaErrorHandler";
+
+/**
+ * Echos what the user type.
+ */
+export const handleEcho: CommandHandler = async (Becca, interaction, t) => {
+  try {
+    interaction.ephemeral = true;
```
Comment (naomi-lgbt) [inline src/commands/subcommands/misc/handleEcho.ts:10]:
```suggestion
    await interaction.deferReply({ ephemeral: true });
```

This should be a deferred reply.

File: src/commands/subcommands/misc/handleEcho.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    const input = interaction.options.getString("input") || "";
```
Comment (naomi-lgbt) [inline src/commands/subcommands/misc/handleEcho.ts:11]:
```suggestion
    const input = interaction.options.getString("input", true);
```

It's a required input, so we call tell the `.getString()` method such and avoid needing a fallback.

File: src/commands/subcommands/misc/handleEcho.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    await interaction.reply({
+      content: input,
+    });
```
Comment (naomi-lgbt) [inline src/commands/subcommands/misc/handleEcho.ts:14]: This isn't quite what we're looking for.

The `input` should be sent as a message to the `interaction.channel` (will need to validate the channel exists and is a text channel).

The response to the interaction should be a confirmation or such.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
We'll also need to validate that the member calling this command has the `Manage Messages` permission.

File: src/commands/subcommands/misc/handleEcho.ts (hunk 2 above)
Comment (a2937) [reply src/commands/subcommands/misc/handleEcho.ts]: Right. Yeah it shouldn't need a fallback.

File: src/commands/subcommands/misc/handleEcho.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+export const handleEcho: CommandHandler = async (Becca, interaction, t) => {
+  try {
+    await interaction.deferReply({ ephemeral: true });
+    const input = interaction.options.getString("input", true);
+    if (
+      interaction.memberPermissions?.has(PermissionFlagsBits.ManageMessages)
+    ) {
+      interaction.channel.send(input);
+    } else {
+      //TODO: Use the translation system to get the text.
```
Comment (naomi-lgbt) [inline src/commands/subcommands/misc/handleEcho.ts:19]: We can use the `responses:noPermission` translation value (we'll need to use the `getRandomValue` util for that)

File: src/commands/subcommands/misc/handleEcho.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+/**
+ * Echos what the user type.
+ */
+export const handleEcho: CommandHandler = async (Becca, interaction, t) => {
+  try {
+    await interaction.deferReply({ ephemeral: true });
+    const input = interaction.options.getString("input", true);
+    if (
+      interaction.memberPermissions?.has(PermissionFlagsBits.ManageMessages)
```
Comment (naomi-lgbt) [inline src/commands/subcommands/misc/handleEcho.ts:15]: A couple of things on this one.

- Let's invert the boolean, such that we can use early return patterns. "If not has permissions, editReply 'cannot use' and return"
- The `interaction` payload is validated by a typeguard, which means you can use `interaction.member.permissions.has` instead - with no optional chaining.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: src/commands/subcommands/misc/handleEcho.ts (hunk 5 above)
Comment (a2937) [reply src/commands/subcommands/misc/handleEcho.ts]: Done.

File: src/commands/subcommands/misc/handleEcho.ts (hunk 4 above)
Comment (a2937) [reply src/commands/subcommands/misc/handleEcho.ts]: Did I do this right?

File: src/commands/subcommands/misc/handleEcho.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    const input = interaction.options.getString("input", true);
+    if (
+      !interaction.member.permissions.has(PermissionFlagsBits.ManageMessages)
+    ) {
+      await interaction.reply({
+        content: tFunctionArrayWrapper(t, "responses:noPermission"),
+      });
+    } else {
+      interaction.channel.send(input);
+    }
```
Comment (naomi-lgbt) [inline src/commands/subcommands/misc/handleEcho.ts:23]:
```suggestion
    if (
      !interaction.member.permissions.has(PermissionFlagsBits.ManageMessages)
    ) {
      await interaction.editReply({
        content: tFunctionArrayWrapper(t, "responses:noPermission"),
      });
      return;
    }
    interaction.channel.send(input);
    await interaction.editReply({
      content: "Your message has been posted.",
    });
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Should be the last nitpick

File: src/commands/subcommands/misc/handleEcho.ts (hunk 4 above)
Comment (naomi-lgbt) [reply src/commands/subcommands/misc/handleEcho.ts]: Should be correct, yes!

**Review by naomi-lgbt — APPROVED** [review]

**Comment by naomi-lgbt** [comment]
## Thank You

Thank you for your continued contribution to our project. We have reviewed your pull request and are happy to accept these changes.

Please continue to watch for issues labelled `help wanted`, as these will be additional opportunities to contribute.

You can also see all open issues [through our contributor tool](https://contribute.naomi.lgbt)

**Pull request merged**